### DISCUSSION TITLE
Understand the uniqBy implementation

### DISCUSSION THREAD

**Discussion by mantey-github** [ramda/ramda#3342] [Q&A]
Hi Ramda team,

From the uniqBy implementation

```
var set = new _Set();
  var result = [];
  var idx = 0;
  var appliedItem, item;

  while (idx < list.length) {
    item = list[idx];
    appliedItem = fn(item);

    if (set.add(appliedItem)) {
      result.push(item);
    }

    idx += 1;
  }

  return result;
```

new Set() was used in order to check the uniqueness of the elements in the list.  
I would like to know why the `add()` method was used and not the `has()`? The condition `if (set.add(appliedItem))`  will always return true or?

**Comment by litemerafrukt** [comment]
Hi @mantey-github!

I'm not on the ramda team but got intrigued by your question :)

At a glance one could think it is a standard javascript Set but it is an internal ramda _Set where the add function returns true/false depending on if the item was added. Look into [_Set.prototype.add](https://github.com/ramda/ramda/blob/c207cf1be650e8373bef85a77198ba23647c9165/source/internal/_Set.js#L14) and [hasOrAdd](https://github.com/ramda/ramda/blob/c207cf1be650e8373bef85a77198ba23647c9165/source/internal/_Set.js#L36)
